lib/passport-wsfed-saml2/saml.js

```diff
@@ -119,7 +124,34 @@ SAML.prototype.validateSignature = function (xml, options, callback) {
   }
 };
 
+SAML.prototype.validateCertExpiration = function (samlAssertion) {
+  // TODO - is this version-dependent?
+  if(samlAssertion.getElementsByTagName("X509Certificate").length > 0){
+    cert = samlAssertion.getElementsByTagName("X509Certificate")[0].textContent;
+  } else {
+    cert = this.options.cert;
+  }
+
+  cert = utils.certToPEM(cert);
+  cert = x509.parseCert(cert)
```

Comment:
Rather than constantly reassigning this variable, it may make more sense to do some pipeline operations on it or just compose the functions together.